inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,192 @@
+contributing to the project. These are guidelines the maintainers would
+like contributors to adhere to, and exist to make the process flow more
+smoothly.
+
+As a contributor you should try to make accepting your code as easy as
+you can, following the contributing docs greatly increases the chance
+your contribution will be accepted.
+
+Some common tidyverse conventions are
+
+  - Add a bullet to `NEWS.md` for each change referencing the issue
+    number and your GitHub username.
+  - Add `Closes #123` at the end of your commit message to automatically
+    close the issue with the PR is merged.
+  - Document functions with
+    [roxygen2](https://cran.r-project.org/web/packages/roxygen2/) and be
+    sure to run `devtools::document()` before submitting.
+
+## Prerequisites
+
+  - For this guide to make sense, you’ll need to be acquainted with Git
+    and GitHub. If you are not, see [Happy Git and GitHub for the
+    useR](http://happygitwithr.com/) by Jenny Bryan.
+
+  - Before you do a pull request, you should always file an issue and
+    make sure someone from the tidyverse team agrees that it’s a
+    problem, and is happy with your basic proposal for fixing it. If
+    you’ve found a bug, first create a minimal
+    [reprex](https://www.tidyverse.org/help/#reprex). We don’t want you
+    to spend a bunch of time on something that we don’t think is a good
+    idea.
+
+There are lots of ways to contribute *other* than by writing code. See
+[Contribute to the tidyverse](https://www.tidyverse.org/contribute/) for
+more on this.
+
+## Making a pull request - overview
+
+  - When in doubt, discuss in an issue before doing lots of work.
+  - Use GitHub to fork and clone the package repo, and a branching
+    workflow to make your changes.
+  - Uphold the design principles and package mechanics outlined below.
+  - Make sure the package still passes `R CMD check` locally for you.
+    It’s a good idea to do that before you touch anything, so you have
+    a ***baseline***.
+  - Match the existing code style. For this purpose, you should follow
+    the tidyverse style guide <http://style.tidyverse.org>.
+  - Update the documentation source, if your PR changes any behavior. We
+    use [**roxygen2**](https://cran.r-project.org/package=roxygen2), so
+    you must edit the roxygen comments above the function; never edit
+    `NAMESPACE` or `.Rd` files by hand.
+  - Do not update the pkgdown-created website, i.e. the files generated
+    below `docs/`.
+      - To update the roxygen documentation *without* changing the
+        pkgdown site, you can run `devtools::document()`.
+
+### Style
+
+For tidyverse projects read the [Style
+Guide](http://style.tidyverse.org) and use the
+[lintr](https://cran.r-project.org/package=lintr) package to find code
+which does not adhere to the style guide.
+
+``` r
+# install.packages("lintr")
+lintr::lint_package()
+```
+
+Remember to include style changes only in code you are contributing.
```

Comment:
I'd say something like: Be careful to only make style changes to the code your are contributing. If there is a lot of code that doesn't meet the style guide, it would be better to file an issue or a PR to fix that first.